avogadro/core/cube.cpp

```diff
@@ -23,21 +24,32 @@ Cube::~Cube()
   m_lock = nullptr;
 }
 
+std::vector<float>::const_iterator Cube::getRowIter(int j, int k) const
+{
+    return m_data.cbegin() + m_points.x() * (k * m_points.y() + j);
+}
+
+
 bool Cube::setLimits(const Vector3& min_, const Vector3& max_,
                      const Vector3i& points)
 {
-  // We can calculate all necessary properties and initialise our data
+  // Calculate the maximum dimension
+  int maxPoints = std::max({points.x(), points.y(), points.z()});
+  Vector3i equalPoints(maxPoints, maxPoints, maxPoints);
+
+  // Recalculate spacing based on equal points
   Vector3 delta = max_ - min_;
-  m_spacing =
-    Vector3(delta.x() / (points.x() - 1), delta.y() / (points.y() - 1),
-            delta.z() / (points.z() - 1));
+  m_spacing = Vector3(delta.x() / (equalPoints.x() - 1),
```

Comment:
Okay, I can recheck and could say here's the issue lies. 

When I was having different dimensions (as a non-uniform grid), lets say `m_points.x() != m_points.y() != m_points.z()` it was showing me a discrete mesh. (looking like the parts of the mesh has been seperated). 

So I thought to use the equal points as a workaround but it leads to the wrong orientation. Any tip for fixing it? @ghutchis

Reply:
Okay, but that doesn't work. Most "cubes" are going to be rectangular solids. They have equal spacing, but different number of points along different axes. Caffeine is a good example here, since it's fairly flat, but kinda a rectangle in the other two dimensions.

CH4 or water aren't great examples because they're roughly cubic.

This code *definitely* shouldn't change. You can't compute the spacing this way because the number of points will be different along different axes.

I see the weird "mesh" when I revert the changes to this code, but that just means the bug is somewhere else.

Reply:
Yes... I got it. Will look into the code again and try to find out the issue tomorrow. Thanks for your reply.